Integration/ClientIntegration.php

```diff
@@ -535,7 +535,9 @@ private function handleException(\Exception $exception)
             if ($exception->getRetry()) {
                 // This type of exception indicates that we can requeue the contact.
                 $this->logIntegrationError($exception, $this->contact);
-                $this->retry = true;
+                // set to Client retry setting and IS an API payload
+                $clientRetryConfig = json_decode($this->contactClient->getApiPayload(), true)['settings']['autoRetry'];
```

Comment:
This will fail if the client doesn't yet have the setting. Checking empty on the next line isn't enough.